src/OrchardCore/OrchardCore.Configuration.KeyVault/Extensions/AzureKeyVaultConfigurationExtension.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using Azure.Core;
+using Azure.Extensions.AspNetCore.Configuration.Secrets;
+using Azure.Identity;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using OrchardCore.Configuration.KeyVault.Services;
+
+namespace OrchardCore.Configuration.KeyVault.Extensions
+{
+    public static class AzureKeyVaultConfigurationExtension
+    {
+        /// <summary>
+        /// Adds Azure Key Vault as a Configuration Source.
+        /// </summary>
+        /// <param name="builder">The host builder to configure.</param>
+        /// <returns>The host builder.</returns>
+        public static IHostBuilder AddOrchardCoreAzureKeyVault(this IHostBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            return builder.AddOrchardCoreAzureKeyVault(new DefaultAzureCredential(includeInteractiveCredentials: true));
+        }
+
+        /// <summary>
+        /// Adds Azure Key Vault as a Configuration Source.
+        /// </summary>
+        /// <param name="builder">The host builder to configure.</param>
+        /// <param name="tokenCredential">The token credential to use for authentication.</param>
+        /// <returns>The host builder.</returns>
+        public static IHostBuilder AddOrchardCoreAzureKeyVault(this IHostBuilder builder, TokenCredential tokenCredential)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.ConfigureAppConfiguration((context, builder) =>
+            {
+                // Here 'configuration' is a manager being a builder, and also an 'IConfigurationRoot'
+                // if get from the 'context', allowing to get values from the providers already added
+                // without having to build a configuration on the fly that would need to be disposed.
+                AddOrchardCoreAzureKeyVault(builder, context.Configuration, tokenCredential);
+            });
+
+            return builder;
+        }
+        /// <summary>
+        /// Adds Azure Key Vault as a Configuration Source.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <returns>The web host builder.</returns>
+        public static IWebHostBuilder AddOrchardCoreAzureKeyVault(this IWebHostBuilder builder)
```

Comment:
Is there a need for `IWebHostBuilder` while we already adding to `IHostBuilder`?

Reply:
Normally only an extension on `IConfigurationBuilder` would have been required but here the singularity is that we need both a builder to add the Key Vault source and also an `IConfiguration` to retrieve some needed data (from the already added providers) to add this source.

Not sure it's good but that's why I kept the one on `IHostBuilder` because it provides a config manager that is both a builder and an `IConfiguration`, and to not break anything.

Not sure the one on `IWebHostBuilder` is necessary but it takes only a few lines of code by using the same pattern that calls the same shared method.

The recommendation now is to use the one on the host builder `Configuration` which is a configuration manager as described above, and that calls the same shared method.

The main goal of this PR being to not build a configuration on the fly, and that was not disposed.